c/include/arrow-adbc/adbc_driver_manager.h

```diff
@@ -31,6 +32,17 @@ extern "C" {
 #ifndef ADBC_DRIVER_MANAGER_H
 #define ADBC_DRIVER_MANAGER_H
 
+typedef uint16_t AdbcLoadOption;
```

Comment:
Given how "option" is overloaded in ADBC already, this could be called `AdbcLoadFlags`.

`uint32_t` is also a better default type for this.